zio/src/main/scala/log/effect/zio/ZioLogWriter.scala

```diff
@@ -78,7 +76,7 @@ object ZioLogWriter {
 
   implicit final private def instance[E]: EffectSuspension[IO[E, ?]] =
     new EffectSuspension[IO[E, ?]] {
-      def suspend[A](a: =>A): IO[E, A] = IO.sync(a)
+      def suspend[A](a: =>A): IO[E, A] = IO.suspend(IO.succeed(a))
```

Comment:
@barambani  ^^

Reply:
to respect the semantic that `suspend` accepts any thunk (side effecting ones) we could go with something like
```scala
implicit final private val instance: EffectSuspension[Task] =
  new EffectSuspension[Task] {
    def suspend[A](a: =>A): Task[A] = IO.effect(a)
  }
```
this means that we have to change also `consoleLogZio` to return a `Task` and [this](https://github.com/laserdisc-io/log-effect/blob/master/zio/src/test/scala/ConsoleLogWriterTest.scala#L11) to 
```scala
private def capturedConsoleOutOf(write: Task[Unit]): String
```